### ISSUE TITLE
Low resolution and low color in images of a PDF converted to PDFA

### ISSUE THREAD

**Issue by luisforra** [gotenberg/gotenberg#939]
Using last version 8,

When converting a pdf to pdfa ou generating directly a pda, the images are in very low resolution, 

Test with this pdf:

https://nlsblog.org/wp-content/uploads/2020/06/image-based-pdf-sample.pdf

`curl \
--request POST https://demo.gotenberg.dev/forms/pdfengines/convert \
--form files=@/home/fluis/image-based-pdf-sample.pdf \
--form pdfa=PDF/A-1b \
--form pdfua=true \
-o pdf_a_img.pdf`

`curl \
--request POST http://192.168.22.52:3000/forms/pdfengines/convert \
--form files=@/home/fluis/image-based-pdf-sample.pdf \
--form pdfa=PDF/A-1b \
--form pdfua=true \
-o pdf_a_img.pdf`

result: https://drive.google.com/file/d/1E3mRrRHdjQumVaHm0YD4jOKxyL8sEYG2/view?usp=sharing

I tried changing the image resolution and lossless encoding but nothing changes.

Thank you for your help

Regards

**Comment by gulien** [comment]
Could you try with the `/forms/libreoffice/convert` endpoint instead? Form fields https://gotenberg.dev/docs/routes#images-libreoffice are not available for /forms/pdfengines/convert`!

**Comment by luisforra** [comment]
Yes, converting with `/forms/libreoffice/convert` works fine.

But if I try do create a PDFA directly from html to PDF I have the same problem, ie:
`
curl \
--request POST http://192.168.22.56:3000/forms/chromium/convert/html \
--form files=@/home/fluis/header.html \
--form files=@/home/fluis/footer.html \
--form files=@/home/fluis/index.html \
--form paperWidth=8.27 \
--form paperHeight=11.7 \
--form marginTop=5cm \
--form marginBottom=2cm \
--form pdfa=PDF/A-1b \
--form pdfua=true \
-o pdf_out_2.pdf 
`

Thank you for your work and help

Regards

**Comment by gulien** [comment]
What form fields are you using to mitigate the issue with `/forms/libreoffice/convert`?

**Comment by luisforra** [comment]
`
curl \

--request POST http://192.168.22.56:3000/forms/libreoffice/convert \

--form files=@/home/fluis/image-based-pdf-sample.pdf \

--form pdfa=PDF/A-1b \

--form pdfua=true \

-o pdf_out_1.pdf
`


I tried with url and has the same problem:
`
curl \
--request POST http://192.168.22.56:3000/forms/chromium/convert/url \

--form url=https://github.com/gotenberg \

--form pdfa=PDF/A-1b \

--form pdfua=true \

-o pdf_out_5.pdf
`

**Comment by luisforra** [comment]
it works, only with in ` /forms/libreoffice/convert`

--form pdfa=PDF/A-1b \
--form pdfua=true \

Thank you

**Comment by luisforra** [comment]
I 'am trying to convert from html or url  directly to PDF/A with `chromium/convert/html` or `chromium/convert/url`  but I get low quality images I tried different sources.
Example:

`
curl
--request POST http://localhost:3000/forms/chromium/convert/url 
--form url=https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/scale 
--form pdfa=PDF/A-1b 
--form pdfua=true 
-o my10.pdf 
`

**Comment by gulien** [comment]
Could you try this scenario and tell me if playing with the « image » form field works in your cass? Thanks 😊 

> Ok, so:
> 
> 1. Convert with `/forms/chromium/convert/url` _without_ using the `pdfa` and `pdfua` form fields*.
> 2. Using the resulting PDF, use `/forms/libreoffice/convert` _with_ the form fields `pdfa` and `pdfua` PLUS play with https://gotenberg.dev/docs/routes#images-libreoffice to see if it mitigates the issue.
> 
> *when you use these form fields, Gotenberg actually uses LibreOffice to convert to PDF/A & PDF/UA.

**Comment by luisforra** [comment]
with this index.html:
```
<html>
<head>
</head>
<body>
  <h1>Title</h1>
  <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/35/Olympic-flag-Victoria.jpg/640px-Olympic-flag-Victoria.jpg">  
</body>
</html>
```
PDF with low resolution image:
```
curl \
--request POST http://localhost:3000/forms/chromium/convert/html \
--form files=@/home/fluis/index.html \
--form pdfa=PDF/A-3b \
--form pdfua=true \
-o nok.pdf 
```

Intermediate PDF with nice image:
```
curl \
--request POST http://localhost:3000/forms/chromium/convert/html \
--form files=@/home/fluis/index.html \
-o intermediate.pdf 
```

Conversion to PDF/A with good image:

```
curl \
--request POST http://localhost:3000/forms/libreoffice/convert \
--form files=@intermediate.pdf \
--form pdfa=PDF/A-3b \
--form pdfua=true \
--form reduceImageResolution=false \
-o ok_1.pdf 
```

or

```
curl \
--request POST http://localhost:3000/forms/libreoffice/convert \
--form files=@intermediate.pdf \
--form pdfa=PDF/A-3b \
--form pdfua=true \
-o ok_2.pdf 
```

**Comment by gulien** [comment]
Thanks! If you update to the latest version (`8.9.0`), `reduceImageResolution` is set to `false` by default, so you will be able to use the Chromium module directly without the intermediate step!

**Comment by luisforra** [comment]
```
curl \
--request POST http://localhost:3000/forms/chromium/convert/html \
--form files=@/home/fluis/index.html \
--form pdfa=PDF/A-3b \
--form pdfua=true \
-o nok.pdf 
```

Creates the PDF with the same problem

log:
```
2024-08-09 20:35:02 Version: 8.9.0
2024-08-09 20:35:02 -------------------------------------------------------
2024-08-09 20:35:02 [SYSTEM] modules: api chromium exiftool libreoffice libreoffice-api libreoffice-pdfengine logging pdfcpu pdfengines pdftk prometheus qpdf webhook 
2024-08-09 20:35:02 [SYSTEM] libreoffice-api: LibreOffice ready to start
2024-08-09 20:35:02 [SYSTEM] chromium: Chromium ready to start
2024-08-09 20:35:02 [SYSTEM] prometheus: collecting metrics
2024-08-09 20:35:02 [SYSTEM] api: server listening on port 3000
2024-08-09 20:35:02 [SYSTEM] pdfengines: exiftool libreoffice-pdfengine pdfcpu pdftk qpdf
2024-08-09 20:42:02 {"level":"info","ts":1723232522.741591,"logger":"api","msg":"request handled","trace":"266d1f43-93ce-45ae-a79a-0dc045449a71","remote_ip":"172.17.0.1","host":"localhost:3000","uri":"/forms/chromium/convert/html","method":"POST","path":"/forms/chromium/convert/html","referer":"","user_agent":"curl/7.81.0","status":200,"latency":2623819399,"latency_human":"2.623819399s","bytes_in":587,"bytes_out":32533}
```

**Comment by gulien** [comment]
@luisforra thanks for the heads up! 

TL;DR: you were in fact.. right! The PDF engine method was not using the same options as the LibreOffice convert method!

**Comment by gulien** [comment]
@luisforra version `8.9.1` should be available within ~1 hour.

**Comment by luisforra** [comment]
Thank you for your help and work, I will give feedback

**Comment by luisforra** [comment]
Is OK now, passes VeraPDF, thank you again

**Comment by luisforra** [comment]
Can you give same example please

**Comment by toooni** [comment]
I've created a PDF twice. Once with 8.3.0 and one in 8.9.0. I've then uploaded it to an online validator on https://www.pdfforge.org/online/en/validate-pdfa. There was one difference:
> Specification: ISO 19005-1:2005, Cause 6.3.5:
For all CIDFont subsets referenced within a conforming file, the font descriptor dictionary shall include a CIDSet stream identifying which CIDs are present in the embedded CIDFont file, as described in PDF Reference Table 5.20

Would it help if I'd upload the the files somewhere?
